### PR TITLE
AtlasEngine: Remove experimental tag and add tracing

### PR DESCRIPTION
**Pull request by lhecker**
With AtlasEngine being enabled by default in 1.16 Preview it would look weird
if the `useAtlasEngine` option would still be called "experimental".
Additionally we're interested in how many users opt out of useAtlasEngine,
indicating major issues that would require us to disable it by default again.

Related to #13936

## Validation Steps Performed
* Toggling `useAtlasEngine` works as expected ✅
* Observe new event with TVPP ✅

### CODE REVIEW

File: doc/cascadia/profiles.schema.json [hunk 1]
```diff
@@ -2237,10 +2237,10 @@
           "description": "Use to set a path to a pixel shader to use with the Terminal. Overrides `experimental.retroTerminalEffect`. This is an experimental feature, and its continued existence is not guaranteed.",
           "type": "string"
         },
-        "experimental.useAtlasEngine": {
-          "description": "Enable using the experimental new rendering engine for this profile. This is an experimental feature, and its continued existence is not guaranteed.",
+        "useAtlasEngine": {
+          "description": "Windows Terminal 1.16 and later ship with a new, performant text renderer. Set this to false to revert back to the old text renderer.",
           "type": "boolean",
-          "default": false
+          "default": true
```
Comment (lhecker) [inline doc/cascadia/profiles.schema.json:2243]: I know that the non-Preview version doesn't have it enabled by default, but I felt like opting in people who modify their settings.json manually via their editor's autocompletion isn't too controversial.

File: src/cascadia/TerminalSettingsEditor/Resources/en-US/Resources.resw [hunk 2]
```diff
@@ -26,36 +26,36 @@
         <value>[base64 mime encoded string representing a byte array form of the .NET Framework object]</value>
         <comment>This is a comment</comment>
     </data>
-                
-    There are any number of "resheader" rows that contain simple 
+
+    There are any number of "resheader" rows that contain simple
```
Comment (DHowett) [inline src/cascadia/TerminalSettingsEditor/Resources/en-US/Resources.resw:30]: if you edited this file outside of Visual Studio, can you re-save it inside VS?

I've complained to them about this before, but... you know.

File: src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp [hunk 3]
```diff
@@ -893,6 +893,20 @@ try
...
+    {
+        const auto& globals = settings->_globals;
+        TraceLoggingWrite(
+            g_hSettingsModelProvider,
+            "AtlasEngine_Enabled",
+            TraceLoggingDescription("Event emitted upon settings load, containing whether one is opted-in/out of useAtlasEngine"),
+            TraceLoggingBool(globals->HasUseAtlasEngine(), "UseAtlasEngineOverridden", "true if useAtlasEngine is set"),
+            TraceLoggingBool(globals->UseAtlasEngine(), "UseAtlasEngine", "true if AtlasEngine is enabled"),
+            TraceLoggingKeyword(MICROSOFT_KEYWORD_MEASURES),
+            TelemetryPrivacyDataTag(PDT_ProductAndServiceUsage));
```
Comment (DHowett) [inline src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp:907]: In team sync you mentioned a bunch of performance data we wanted to get insights into as well. Atlas resizing, etc.
Did you want to include those in this PR as well?

Comment (lhecker) [reply src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp:907]: I'd have to figure out how to implement them first and I felt like getting this crucial change in ASAP is much more important than that.

Comment (zadjii-msft) [reply src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp:907]: I agree with Leonard on this one. Let's start with simple

**Review by DHowett — APPROVED** [review]
I'm cool with this, presuming that atlas is a "compatibility" option

**Comment by DHowett** [comment]
I'm okay with the setting moving in the UI to _Rendering_ where it belongs. I do think its logical place makes no sense today. However, its physical place can remain in `profiles.defaults.useAtlasEngine`. What do you think?

**Comment by DHowett** [comment]
The thing about having a different logical place is that we could have _multiple_ logical places for it. the Rendering page can set the defaults one, but what if you have a profile that overrides it? I don't know.

**Comment by lhecker** [comment]
The latest commits removes the "compatibility" prefix and allows the setting to be changed in both places (rendering and advanced default profile settings).

**Review by DHowett — APPROVED** [review]
thanks so much

File: src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp [hunk 4]
```diff
@@ -893,6 +893,25 @@ try
...
+        {
+            enabled[profile.UseAtlasEngine()]++;
+        }
+
+        TraceLoggingWrite(
+            g_hSettingsModelProvider,
+            "AtlasEngine_Usage",
+            TraceLoggingDescription("Event emitted upon settings load, containing the number of profiles opted-in/out of useAtlasEngine"),
+            TraceLoggingUIntPtr(enabled[0], "UseAtlasEngineDisabled", "Number of profiles for which AtlasEngine is disabled"),
+            TraceLoggingUIntPtr(enabled[1], "UseAtlasEngineEnabled", "Number of profiles for which AtlasEngine is enabled"),
```
Comment (zadjii-msft) [inline src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp:910]: Do we maybe want to break this down further?
* Users that don't have the setting at all (`!HasUseAtlasEngineDisabled()`)
* Users that explicitly opted in
* Users that explicitly opted out

and maybe a total number of profiles too? Trying to figure out how many folks actually just set this in `profiles.defaults` might complicate this...

Comment (DHowett) [reply src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp:910]: Ah right, we can infer that the user opted out if Disabled > 0. The version **is** part of the data we automatically get.

**Review by PankajBhojwani — APPROVED** [review]

File: src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp (hunk 4 above)
Comment (zadjii-msft) [reply src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp:910]: Oh so I guess (Num Preview users)-(preview users with it disabled)=(preview users with it enabled, either by default or explicitly). Okay, that makes sense.

**Review by zadjii-msft — APPROVED** [review]

**Pull request merged**